internal/overlord/servstate/manager.go

```diff
@@ -300,6 +300,30 @@ func (m *ServiceManager) Services(names []string) ([]*ServiceInfo, error) {
 	return services, nil
 }
 
+// StopTimeout returns the worst case duration that will have to be waited for
+// to have all services in this manager stopped.
+func (m *ServiceManager) StopTimeout() time.Duration {
+	m.servicesLock.Lock()
+	defer m.servicesLock.Unlock()
+
+	maxDuration := killDelayDefault
+	for _, service := range m.services {
+		if service == nil {
+			continue
+		}
+		switch service.state {
+		case stateStarting, stateRunning, stateTerminating:
+			if service.killDelay() > maxDuration {
+				maxDuration = service.killDelay()
+			}
+		}
+	}
+
+	// We add a little extra time here to allow for signals to be sent and
+	// processed.
+	return maxDuration + 100*time.Millisecond
```

Comment:
We need to add failDelay here, as that's the amount of time we're willing to wait before the service is supposed to actually be terminated. Otherwise, 100 milliseconds is way too short and may generate bugs due to the process dying while it's still attempting to deliver the final KILL signals.

Reply:
Thanks good pick up!